src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarEntry.cs

```diff
@@ -349,11 +349,17 @@ internal Task ExtractRelativeToDirectoryAsync(string destinationDirectoryPath, b
                     throw new InvalidDataException(SR.TarEntryHardLinkOrSymlinkLinkNameEmpty);
                 }
 
-                linkTargetPath = GetSanitizedFullPath(destinationDirectoryPath, LinkName);
+                linkTargetPath = GetSanitizedFullPath(destinationDirectoryPath,
+                    Path.IsPathFullyQualified(LinkName) ? LinkName : Path.Join(Path.GetDirectoryName(fileDestinationPath), LinkName));
+
                 if (linkTargetPath == null)
                 {
                     throw new IOException(SR.Format(SR.TarExtractingResultsLinkOutside, LinkName, destinationDirectoryPath));
                 }
+
+                // after TarExtractingResultsLinkOutside validation, preserve the original
+                // symlink target path (to match behavior of other utilities).
+                linkTargetPath = LinkName;
```

Comment:
How are other symlink cases able to preserve the correct link target (relative or absolute)?

Reply:
AFAICT, this code is used by all symlink normalizations.

Reply:
Ah, so it appears that prior to your PR, we were always extracting symlinks with an absolute target. As part of the outside-of-directory-bug, links with a relative target would be created with an incorrect target, even inside-of-extracted-directory. 

if `LinkName` is `./file,txt`, the symlink will be created pointing to `destinationDirectory/file.txt`, regardless of where the link was in the directory tree. This change is also fixing that.

Reply:
Probably `GetDestinationAndLinkPaths` doesn't even need to return a path for the link target, as we can directly use `LinkName` when creating the link.

Reply:
We need to call `GetDestinationAndLinkPaths` for validation.

Reply:
I mean:
```
private (string, string?) GetDestinationAndLinkPaths(string destinationDirectoryPath)
```
can probably changed to:
```
private string GetDestinationAndLinkPaths(string destinationDirectoryPath)
```
as we shouldn't use the second item of the tuple (and can directly use `LinkName` instead).

Reply:
Then we will be checking 'is it a link' condition again at callsite (we already check it here for validation).